libraries/SPI/SPI.h

```diff
@@ -54,6 +60,7 @@ class SPIClass {
 public:
   SPIClass();
   void begin();
+  void begin(uint8_t pinSet, uint8_t miso = 0, uint8_t mosi = 0, uint8_t sck = 0, uint8_t ss = 0);
```

Comment:
I would rather suggest keeping just miso/miso/sck/ss arguments. You can determine _pinSet value internally.

Another point, do Arduino libraries typically call SPI.begin internally? If so, maybe adding an independent .pins method would be a more portable choice. That way, one could first call SPI.pins in the sketch and then call .begin method of another library which would do SPI.begin internally.